File: VB/SelectedObjects.Module/MyConfirmationController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.SystemModule

Namespace SelectedObjects.Module
	Public Class MyConfirmationController
		Inherits ViewController(Of DetailView)
		Private defaultMessage As String
		Protected Overrides Sub OnFrameAssigned()
			MyBase.OnFrameAssigned()
			TargetObjectType = GetType(Contact)
			defaultMessage = Frame.GetController(Of DeleteObjectsViewController)().DeleteAction.GetFormattedConfirmationMessage()
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			AddHandler View.CurrentObjectChanged, AddressOf View_CurrentObjectChanged
			View_CurrentObjectChanged(View, New System.EventArgs())
		End Sub
		Private Sub View_CurrentObjectChanged(ByVal sender As Object, ByVal e As System.EventArgs)
			Dim disableDeleteConfirmation As Boolean = String.IsNullOrEmpty((CType(View.CurrentObject, Contact)).Notes)
			If disableDeleteConfirmation Then
				Frame.GetController(Of DeleteObjectsViewController)().DeleteAction.ConfirmationMessage = ""
			Else
				Frame.GetController(Of DeleteObjectsViewController)().DeleteAction.ConfirmationMessage = defaultMessage
			End If
		End Sub
	End Class
End Namespace

File: CS/SelectedObjects.Module/MyConfirmationController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.SystemModule;

namespace SelectedObjects.Module {
    public class MyConfirmationController : ViewController<DetailView> {
        string defaultMessage;
        protected override void OnFrameAssigned() {
            base.OnFrameAssigned();
            TargetObjectType = typeof(Contact);
            defaultMessage = Frame.GetController<DeleteObjectsViewController>().DeleteAction.GetFormattedConfirmationMessage();
        }
        protected override void OnActivated() {
            base.OnActivated();            
            View.CurrentObjectChanged += View_CurrentObjectChanged;
            View_CurrentObjectChanged(View, new System.EventArgs());
        }
        void View_CurrentObjectChanged(object sender, System.EventArgs e) {
            bool disableDeleteConfirmation = string.IsNullOrEmpty(((Contact)View.CurrentObject).Notes);
            Frame.GetController<DeleteObjectsViewController>().DeleteAction.ConfirmationMessage = disableDeleteConfirmation ? "" : defaultMessage;
        }
    }
}